### ISSUE TITLE
Add debug code to output before code is run (online compiler)

### ISSUE THREAD

**Issue by willsherwood** [isaacg1/pyth#95]
while debugging code, it is useful to have the python code printed to the output before the program runs. If the program hangs, it's very hard to debug because the code is never shown.

**Comment by jakobkogler** [comment]
I'll guess you mean the Online Compiler, since offline compiler already does this. Not sure, if this is easy to implement. 

Maybe the online compiler could simple terminate after running for 5 seconds and return the result, as far as it is computed. This would also prevent the server from crashing.

**Comment by willsherwood** [comment]
Yes, this is only an issue for people using the pyth online compiler for example on their phones

**Comment by gcq** [comment]
There is a very easy linux-only way to do the timeout using the [signal](https://docs.python.org/3.4/library/signal.html) module in the standard library. I don't think threads can be avoided if it has to be multiplataform.

**Comment by isaacg1** [comment]
One way to implement this without threads is with a command line flag to set the timeout, that would be enabled in the online versions.

**Comment by Pietu1998** [comment]
When testing programs that seem to hang one can now (since 99838ee14d2b64692edac60c5952bcfabcffc0e0) add `.q` to the start of the program. I would like to see a timeout function, though.

**Comment by Maltysen** [comment]
A fix for this, that covers much more, is to use web sockets to do real time communication from the server to the client. However, this is very involved.
